central/graphql/resolvers/plotted_vulnerabilities_common.go

```diff
@@ -0,0 +1,63 @@
+package resolvers
+
+import (
+	"context"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/search"
+)
+
+func getPlottedVulnsIdsAndFixableCount(ctx context.Context, root *Resolver, args RawQuery) ([]string, int, error) {
+	q, err := args.AsV1QueryOrEmpty()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	q = tryUnsuppressedQuery(q)
+	q.Pagination = &v1.QueryPagination{
+		SortOptions: []*v1.QuerySortOption{
+			{
+				Field:    search.CVSS.String(),
+				Reversed: true,
+			},
+		},
+	}
+	all, err := root.CVEDataStore.Search(ctx, q)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	fixable, err := root.CVEDataStore.Count(ctx,
+		search.ConjunctionQuery(q, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery()))
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return search.ResultsToIDs(all), fixable, nil
+}
+
+func unwrappedPlottedVulnerabilities(ctx context.Context, resolver *Resolver, cveIds []string, args PaginatedQuery) ([]*cVEResolver, error) {
```

Comment:
@charmik-redhat , this seems redundant. Check if `unwrappedVulnerabilitiesV2` serves the purpose.

Reply:
This function is used by `NodeVulnerabilities` sub-resolver inside `PlottedNodeVulnerability` type. It is different from `unwrappedVulnerabilitiesV2` in that it does not query the `CVEDatastore` based on any `field=value` type query, it already has the relevant CVE IDs and just fetches the corresponding CVEs and then applies pagination. CVE IDs are pre-populated in the in the init function  `newPlottedNodeVulnerabilitiesResolver` resolver and used by the sub-resolvers.

In order to use `unwrappedVulnerabilitiesV2` here, we would need to store the `RawQuery` for the plotted Vulns in the `PlottedNodeVulnerablityResolver` struct and query the `CVEDatastore` based on that `RawQuery` for each sub-resolver.